client/source/client.cpp

```diff
@@ -0,0 +1,610 @@
+
+#include "client.h"
+
+Client::Client(const std::string &serverAddress, int port)
+    : serverAddress(serverAddress), port(port), clientSocket(0)
+{
+    memset(&address, 0, sizeof(address));
+    createSocket();
+    connectToServer();
+}
+
+Client::~Client()
+{
+    close(clientSocket);
+}
+
+void Client::createSocket()
+{
+    if ((clientSocket = socket(AF_INET, SOCK_STREAM, 0)) < 0)
+    {
+        perror("Socket creation failed");
+        exit(EXIT_FAILURE);
+    }
+}
+
+void Client::connectToServer()
+{
+    address.sin_family = AF_INET;
+    address.sin_port = htons(port);
+
+    if (inet_pton(AF_INET, serverAddress.c_str(), &address.sin_addr) <= 0)
+    {
+        perror("Invalid address/Address not supported");
+        exit(EXIT_FAILURE);
+    }
+
+    if (connect(clientSocket, (struct sockaddr *)&address, sizeof(address)) < 0)
+    {
+        perror("Connection failed");
+        exit(EXIT_FAILURE);
+    }
+}
+
+void Client::sendFeedback()
+{
+    int foodItemId;
+    int rating;
+    std::string employeeId;
+    std::string comment;
+
+    std::cout << "Enter Food Item ID: ";
+    std::cin >> foodItemId;
+    std::cin.ignore();
+    std::cout << "Enter Employee ID: ";
+    std::getline(std::cin, employeeId);
+
+    std::cout << "Enter Rating(0-5): ";
+    std::cin >> rating;
+    std::cin.ignore();
+
+    std::cout << "Enter Comment: ";
+    std::getline(std::cin, comment);
+    std::string message = "ADD_FEEDBACK:" + employeeId + ":" + std::to_string(foodItemId) + ":" + std::to_string(rating) + ":" + comment;
+    send(clientSocket, message.c_str(), message.length(), 0);
+    std::cout << "Feedback sent\n";
+
+    char buffer[bufferSize] = {0};
+    read(clientSocket, buffer, bufferSize);
+    std::cout << "Response from server: " << buffer << std::endl;
+}
+void Client::viewNotificationsRequestToServer(const std::string &message)
+{
+    send(clientSocket, message.c_str(), message.length(), 0);
+    std::cout << "Notifications request sent\n";
+
+    char buffer[bufferSize] = {0};
+    read(clientSocket, buffer, bufferSize);
+    std::cout << "Notifications:\n" << buffer << std::endl;
+}
+
+void Client::getRecommendations(std::string message)
```

Comment:
Look at creating smaller functions. If they are big, check if we can break them down into smaller ones.